### ISSUE TITLE
[bug]: When creating a new plugin (javascript) Strapi tries to look for a dist directory when there shouldn't be one

### ISSUE THREAD

**Issue by derrickmehaffy** [strapi/sdk-plugin#33]
### What version of `@strapi/blocks-react-renderer` are you using?

```
$ strapi report --all
Launched In: 324 ms
Environment: development
OS: linux-x64
Strapi Version: 5.0.0-rc.8
Node/Yarn Version: yarn/1.22.22 npm/? node/v20.11.1 linux x64
Edition: Community
Database: sqlite
UUID: be5a4b93-531e-4dd9-8b6f-0ed0dc911441
Dependencies: {
  "@strapi/strapi": "5.0.0-rc.8",
  "@strapi/plugin-users-permissions": "5.0.0-rc.8",
  "@strapi/plugin-cloud": "5.0.0-rc.8",
  "react": "^18.0.0",
  "react-dom": "^18.0.0",
  "react-router-dom": "^6.0.0",
  "styled-components": "^6.0.0",
  "better-sqlite3": "9.4.3"
}
Dev Dependencies: {}
[2024-08-05 09:47:13.517] info: Shutting down Strapi
[2024-08-05 09:47:13.520] info: Strapi has been shut down
Done in 1.80s.
```

### What's Wrong?

When creating a new javascript plugin (and installing plugin modules + configuring plugin in the project Strapi will try to load the plugin looking for `./dist/server` directory when one should not exist since it's javascript, not typescript.

![image](https://github.com/user-attachments/assets/1ac0649a-88fd-4c20-b961-7e44f5b9cb6c)

### To Reproduce

- Create a new project
- Create a new plugin
- Move plugin to proper location if it wasn't created there (https://github.com/strapi/sdk-plugin/issues/22)
- Install Plugin modules (https://github.com/strapi/sdk-plugin/issues/34)
- Configure plugin in `./src/plugins.js`
- Try to run Strapi application
- See Error

### Expected Behaviour

Strapi should detect if it's a typescript or javascript project and load the plugin as needed if it's javascript

**Comment by derrickmehaffy** [comment]
If that's what should be done, does the default build command in new packages use packup?

**Comment by innerdvations** [comment]
Yeah, you just have to run `yarn build` in the plugin directory. If you do that, it gets the [same error as Typescript projects](https://github.com/strapi/sdk-plugin/issues/35) which is what I'm working on today.

**Comment by innerdvations** [comment]
Ok, I've done a bit more research and it doesn't seem there's any real reason to build it, so I'm not exactly sure why it's doing that. I'll experiment with just loading it directly.

**Comment by innerdvations** [comment]
I've confirmed that even javascript plugins need to be built, both for the admin and to ensure they build for both ESM and commonjs.

However, javascript plugins were not building the server correctly, which I've fixed in #38
